src/schema/artwork/contextGrids/index.ts

```diff
@@ -0,0 +1,169 @@
+          type: new GraphQLList(GraphQLString),
+          description:
+            "List of artwork IDs to exclude from the response (irrespective of size)",
+        },
+      }),
+    },
+  }),
+  resolveType,
+})
+
+export const ContextGrids: GraphQLFieldConfig<any, ResolverContext> = {
+  type: new GraphQLList(ContextGridType),
+  resolve: async (
+    artwork,
+    _options,
+    { saleLoader, relatedFairsLoader, relatedShowsLoader }
+  ) => {
+    const { id, artist, partner, sale_ids } = artwork
+
+    // If the artwork is in an auction, return a context that includes the auction
+    if (sale_ids && sale_ids.length > 0) {
+      const sale = await saleLoader(sale_ids[0])
+      if (sale && sale.is_auction) {
+        if (sale.auction_state === "closed") {
+          return [
+            ...(artist
+              ? [{ gridType: "ArtistArtworkGrid", artist, artwork }]
+              : []),
+            {
+              gridType: "RelatedArtworkGrid",
+              artist,
+              artwork,
+            },
+          ]
+        } else {
+          return [
+            {
+              gridType: "AuctionArtworkGrid",
+              sale,
+              artwork,
+            },
+          ]
+        }
+      }
+    }
+
+    const fairs = await relatedFairsLoader({ artwork: [id], size: 1 })
+    const fair = first(fairs)
+    if (fair && fair.has_full_feature) {
+      const relatedFairShowsResponse = await relatedShowsLoader({
+        artwork: [id],
+        at_a_fair: true,
+      })
+
+      // If the artwork is in a fair, return a context that includes that fair show
+      if (
+        relatedFairShowsResponse &&
+        relatedFairShowsResponse.body &&
+        relatedFairShowsResponse.body.length > 0
+      ) {
+        const fairShow = first(relatedFairShowsResponse.body)
+        if (fairShow) {
+          return [
+            {
+              gridType: "ShowArtworkGrid",
+              show: fairShow,
+              atAFair: true,
+              artwork,
+            },
+            ...(artist
+              ? [{ gridType: "ArtistArtworkGrid", artist, artwork }]
+              : []),
+            { gridType: "RelatedArtworkGrid", artwork },
+          ]
+        }
+      }
+    }
+
+    const relatedShowsResponse = await relatedShowsLoader({
+      artwork: [id],
+      size: 1,
+      active: false,
+      at_a_fair: false,
+    })
+
+    // If the artwork is in a show, return a context that includes that show
+    if (
+      relatedShowsResponse &&
+      relatedShowsResponse.body &&
+      relatedShowsResponse.body.length > 0
+    ) {
+      const show = first(relatedShowsResponse.body)
+      return [
+        { gridType: "ShowArtworkGrid", show, artwork },
+        ...(artist ? [{ gridType: "ArtistArtworkGrid", artist, artwork }] : []),
+        ...(partner
+          ? [{ gridType: "PartnerArtworkGrid", partner, artwork }]
+          : []),
+        { gridType: "RelatedArtworkGrid", artwork },
```

Comment:
We would need to fetch for `layer` here in order to decide whether or not to render this.

In any case, clients will need to exclude grids which have no `artworks`.